packages/common/src/content/_internal/internalContentUtils.ts

```diff
@@ -95,6 +96,47 @@ export const getContentBoundary = (item: IItem): IHubGeography => {
   return boundary;
 };
 
+/**
+ * Constructs IExtent from numeric item extent array
+ * @param extent Raw item extent array
+ * @returns IExtent
+ */
+const getExtentObject = (extent: number[][]): IExtent => {
+  return isBBox(extent)
+    ? ({ ...bBoxToExtent(extent), type: "extent" } as unknown as IExtent)
+    : undefined;
+};
+
+/**
+ * Derives proper IHubLocation given an ArcGIS Item.  If no
+ * location (item.properties.location) is present, one will be
+ * constructed from the item's extent.
+ * If item.properties.boundary === 'none', this function will strip
+ * location and return { type: 'none' } per this document:
+ * https://confluencewikidev.esri.com/display/Hub/Hub+Location+Management
+ * @param item ArcGIS Item
+ * @returns IHubLocation
+ */
+export const deriveLocationFromItem = (item: IItem): IHubLocation => {
+  const { properties, extent } = item;
+  if (properties?.boundary === "none") {
+    return { type: "none" };
+  }
```

Comment:
I know you just copied @abp6318's code here, but it seems odd that we would favor the deprecated  `boundary` property. I assume that the code that sets `location` deletes `boundary`, which is fine and good if that's the case, but I can't think of a reason that we'd open ourselves up to the possibility of having `boundary` override `location` and I dont' see anything in https://confluencewikidev.esri.com/display/Hub/Hub+Location+Management that explains why we would do this.

Reply:
Would you suggest removing this logic all-together?  I'm in favor of dropping it.

Reply:
I see what happened, this fn wasn't just copied over verbatim. We need to update the logic here to early return location if it's present _first_. then early return if boundary is none, then finally derive from extent.